dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ProgramEncoder.java

```diff
@@ -63,13 +64,38 @@ public static ProgramEncoder withContext(EncodingContext context) throws Invalid
 
     public BooleanFormula encodeFullProgram() {
         return context.getBooleanFormulaManager().and(
+                encodeNondeterministicBounds(),
                 encodeMemory(),
                 encodeControlFlow(),
                 encodeFinalRegisterValues(),
                 encodeFilter(),
                 encodeDependencies());
     }
 
+    public BooleanFormula encodeNondeterministicBounds() {
```

Comment:
Wouldn't it be more appropriate to name it `encodeProgramConstants`, not matter if they are non-deterministic or not?

Reply:
You could argue that, since the method is public and thus may abstract from what its implementation actually does.  On the other hand, if encoding constants would include something else or different, i.e. sets of distinct constants, a new `encodeProgramConstants` could also be appropriate.

Reply:
Well yes, but so far you use `INonDet` interchangably with the notion of constants.
If you had classes like `NonDet` and `Constant` then I would totally agree that you can have two different methods.
Right now, the situation is that "constants" get created, but "nondeterministic values" get encoded, i.e., there is a sort of implicit conversion from constants to nondets going on.
I think this is unnecessarily confusing (for now).